### PR TITLE
Dev aicoe followups bugfixes

### PR DESCRIPTION
**Pull request by shivrajp1300**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Dates issue fix
Added drafts to shared pref
other bug fixes

Fixes [link to issue]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt [hunk 1]
```diff
@@ -68,7 +68,7 @@ class NetworkModule {
           level =
             if (BuildConfig.DEBUG) {
               HttpLoggingInterceptor.Level.BODY
-            } else HttpLoggingInterceptor.Level.BASIC
+            } else HttpLoggingInterceptor.Level.BODY
```
Comment (farmaan-e) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt:71]: Remove body logging since image bytes will also be logged

File: android/quest/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <application
         android:name=".QuestApplication"
         android:allowBackup="true"
+        android:debuggable="true"
```
Comment (farmaan-e) [inline android/quest/src/main/AndroidManifest.xml:11]: Remove debuggable

File: android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt [hunk 3]
```diff
@@ -60,7 +60,8 @@ class QuestApplication : OpenSrpApplication(), DataCaptureConfig.Provider, Confi
     if (BuildConfig.DEBUG) {
       Timber.plant(Timber.DebugTree())
     } else {
-      Timber.plant(ReleaseTree())
+      Timber.plant(Timber.DebugTree())
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt:63]: Revert back to release tree

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 4]
```diff
@@ -625,19 +615,42 @@ constructor(
    */
   fun saveDraftQuestionnaire(questionnaireResponse: QuestionnaireResponse) {
     viewModelScope.launch {
-      val questionnaireHasAnswer =
-        questionnaireResponse.item.any{it.item.get(1).hasAnswer()}
+      val questionnaireHasAnswer = questionnaireResponse.item.any { it.item.get(1).hasAnswer() }
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:618]: Please have proper null check here

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 5]
```diff
@@ -625,19 +615,42 @@ constructor(
...
-        _isDraftSaved.postValue(true)
-      }else{
-        _isDraftSaved.postValue(true)
+        try {
+          val flwId = getUserName()
+          val ref = Reference()
+          ref.reference = "Practitioner/$flwId"
+          // set author
+          questionnaireResponse.author = ref
+          questionnaireResponse.id = "qr-${Random.nextBits(1000000)}"
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:626]: Just use UUID to generate id

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterScreen.kt [hunk 6]
```diff
@@ -910,7 +912,13 @@ fun SyncedPatientCard(patientData: Patient, patient: RegisterViewModel.AllPatien
 
         Row(modifier = Modifier.padding(vertical = 4.dp)) {
           Box(modifier = Modifier.padding(vertical = 4.dp, horizontal = 36.dp)) {
-            Text(text = "Visited ${convertToDate(patient.meta.lastUpdated)}")
+
+            if (patient.patient?.extension?.isNotEmpty() == true){
+              Text(text = "Visited ${patient.patient?.extension?.get(0)?.value?.asStringValue()
+                ?.let { convertToDateStringFromString(it)}}")
+            }else{
+              Text(text = "Visited ${convertToDate(patient.meta.lastUpdated)}")
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterScreen.kt:920]: Use the patient registration extension as this

```json
{
"extension": [
    {
      "url": "https://midas.iisc.ac.in/fhir/StructureDefinition/patient-registraion-date",
      "valueDateTime": "2024-08-11T01:37:00+05:30"
    }
  ]

}
```

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt [hunk 7]
```diff
@@ -709,13 +704,49 @@ constructor(
         // ... your search criteria
       }.map {
         it.resource
-      }.sortedByDescending { it.meta.lastUpdated }
+      }.filter { patient ->
+        (patient?.generalPractitioner?.firstOrNull()?.reference?.toString()?.substringAfter("/") ?: "").equals(getUserName(), true)
+      }.sortedByDescending {
+        it.meta.lastUpdated
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt:710]: Use the patient registration date

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt [hunk 8]
```diff
@@ -786,31 +814,64 @@ constructor(
...
+        val draftResponsesJson = sharedPreferencesHelper.read<String>(SharedPreferenceKey.DRAFTS.name, false)
+        if (!draftResponsesJson.isNullOrEmpty()){
+          val allDrafts = parser?.parseResource(draftResponsesJson) as Bundle
+
+          allDrafts?.entry?.forEach {
+            allResponses.add(it.resource as QuestionnaireResponse)
+          }
+        }
+      }catch (e: Exception){
+        e.printStackTrace()
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt:872]: Use Timber

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt [hunk 9]
```diff
@@ -835,14 +896,37 @@ constructor(
...
+          }
+
+        }
+
+        // Save the updated bundle
+        val bundleJson = parser.encodeResourceToString(allDrafts)
+        sharedPreferencesHelper.write<String>(SharedPreferenceKey.DRAFTS.name, bundleJson)
+
+      }catch (e: Exception){
+        e.printStackTrace()
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt:925]: Use Timber

File: android/quest/src/main/java/org/smartregister/fhircore/quest/util/OpensrpDateUtils.kt [hunk 10]
```diff
@@ -26,4 +26,18 @@ object OpensrpDateUtils {
...
     }
+
+    fun convertToDateStringToDate(input: String): Date {
+        val format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX") // Adjust format if necessary
+        val outputFormat = SimpleDateFormat("dd MMMM yyyy HH:mm", Locale.getDefault())
+
+        val dateObj: Date? = try {
+            format.parse(input)
+        } catch (e: Exception) {
+            e.printStackTrace()
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/util/OpensrpDateUtils.kt:37]: Use Timber

**Review by farmaan-e — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt (hunk 1 above)
Comment (farmaan-e) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt]: Not resolved

File: android/quest/src/main/AndroidManifest.xml (hunk 2 above)
Comment (farmaan-e) [reply android/quest/src/main/AndroidManifest.xml]: Unresolved

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 11]
```diff
@@ -625,19 +615,42 @@ constructor(
...
+          }
+
+          // Save the updated bundle
+          val bundleJson = parser.encodeResourceToString(draftResBundle)
+          sharedPreferencesHelper.write<String>(SharedPreferenceKey.DRAFTS.name, bundleJson)
+
+          _isDraftSaved.postValue(true)
+        }catch (e: Exception){
+          Timber.d(
+            "saveDraftQuestionnaire: ${e.printStackTrace()}")
```
Comment (farmaan-e) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt]: Remove stack trace. Log as error and not a debug

**Review by farmaan-e — APPROVED** [review]
